### ISSUE TITLE
Another System.NullReferenceException && System.InvalidCastException

### ISSUE THREAD

**Issue by PatrikSamuelTauchim** [ZaneDubya/UltimaXNA#268]
Something went wrong... :-(

Again, group of NPCs was fighting and getting killed while client crashed

```
System.NullReferenceException was unhandled
  HResult=-2147467261
  Message=Object reference not set to an instance of an object.
  Source=UltimaXNA
  StackTrace:
       at UltimaXNA.Ultima.World.EntityViews.CorpseView.Draw(SpriteBatch3D spriteBatch, Vector3 drawPosition, MouseOverList mouseOverList, Map map) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\EntityViews\CorpseView.cs:line 32
       at UltimaXNA.Ultima.World.WorldViews.IsometricRenderer.DrawEntities(Map map, Position3D center, MousePicking mousePicking, Vector2& renderOffset) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\WorldViews\IsometricView.cs:line 200
       at UltimaXNA.Ultima.World.WorldViews.IsometricRenderer.Update(Map map, Position3D center, MousePicking mousePick) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\WorldViews\IsometricView.cs:line 84
       at UltimaXNA.Ultima.World.WorldView.Draw(Double frameTime) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\WorldView.cs:line 119
       at UltimaXNA.UltimaGame.Draw(GameTime gameTime) in C:\Users\VM\Desktop\UltimaXNA-master\dev\UltimaGame.cs:line 224
       at Microsoft.Xna.Framework.Game.DrawFrame()
       at Microsoft.Xna.Framework.Game.Tick()
       at Microsoft.Xna.Framework.Game.HostIdle(Object sender, EventArgs e)
       at Microsoft.Xna.Framework.GameHost.OnIdle()
       at Microsoft.Xna.Framework.WindowsGameHost.RunOneFrame()
       at Microsoft.Xna.Framework.WindowsGameHost.ApplicationIdle(Object sender, EventArgs e)
       at System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
       at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.Run(Form mainForm)
       at Microsoft.Xna.Framework.WindowsGameHost.Run()
       at Microsoft.Xna.Framework.Game.RunGame(Boolean useBlockingRun)
       at Microsoft.Xna.Framework.Game.Run()
       at UltimaXNA.Bootstrapper.StartEngine() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 79
       at UltimaXNA.Bootstrapper.Initialize() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 62
       at UltimaXNA.Bootstrapper.Main(String[] args) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 29
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 

```

When I tryed to login or go to affected are I got different one:

```
System.InvalidCastException was unhandled
  HResult=-2147467262
  Message=Unable to cast object of type 'UltimaXNA.Ultima.World.Entities.Items.Item' to type 'UltimaXNA.Ultima.World.Entities.Items.Containers.Container'.
  Source=UltimaXNA
  StackTrace:
       at UltimaXNA.Ultima.World.Managers.EntityManager.GetObject[T](Serial serial, Boolean create) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\Managers\EntityManager.cs:line 126
       at UltimaXNA.Ultima.World.WorldClient.ReceiveAddMultipleItemsToContainer(IRecvPacket packet) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\World\WorldClient.cs:line 295
       at UltimaXNA.Core.Network.NetworkClient.InvokeHandlers(List`1 packetHandlers, Byte[] buffer, Int32 length) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Core\Network\NetworkClient.cs:line 539
       at UltimaXNA.Core.Network.NetworkClient.Slice() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Core\Network\NetworkClient.cs:line 480
       at UltimaXNA.UltimaGame.Update(GameTime gameTime) in C:\Users\VM\Desktop\UltimaXNA-master\dev\UltimaGame.cs:line 203
       at Microsoft.Xna.Framework.Game.Tick()
       at Microsoft.Xna.Framework.Game.HostIdle(Object sender, EventArgs e)
       at Microsoft.Xna.Framework.GameHost.OnIdle()
       at Microsoft.Xna.Framework.WindowsGameHost.RunOneFrame()
       at Microsoft.Xna.Framework.WindowsGameHost.ApplicationIdle(Object sender, EventArgs e)
       at System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
       at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.Run(Form mainForm)
       at Microsoft.Xna.Framework.WindowsGameHost.Run()
       at Microsoft.Xna.Framework.Game.RunGame(Boolean useBlockingRun)
       at Microsoft.Xna.Framework.Game.Run()
       at UltimaXNA.Bootstrapper.StartEngine() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 79
       at UltimaXNA.Bootstrapper.Initialize() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 62
       at UltimaXNA.Bootstrapper.Main(String[] args) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 29
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
```

:-(

**Comment by ZaneDubya** [comment]
@Eskymak Try after the latest set of commits? 74daa9ca0c03d9fd0dbb23e9df840418d6fbed73 might fix this.

**Comment by PatrikSamuelTauchim** [comment]
@ZaneDubya Looks fine, spawned lot of dummy npcs. even mages with spells and this kind of "effective" stuff, made epic battle and no crash.

Just got #265 so I will try to figure it out. Probably related to something that city guards do.

**Comment by ZaneDubya** [comment]
Excellent, closing as addressed by 74daa9ca0c03d9fd0dbb23e9df840418d6fbed73
